raiden/api/validations/channel_validator.py

```diff
@@ -118,3 +119,37 @@ def can_close_channel(token_address,
         )
 
         return channels_to_close
+
+    @staticmethod
+    def validate_and_get_channels_to_settle(token_address: TokenAddress,
+                                            creator_address: Address,
+                                            partner_address: Address,
+                                            registry_address: Address,
+                                            chain_state: ChainState):
+
+        if not is_binary_address(token_address):
+            raise InvalidAddress("Expected binary address format for token in channel settle")
+
+        if not is_binary_address(partner_address):
+            raise InvalidAddress("Expected binary address format for partner address in channel settle")
+
+        valid_tokens = views.get_token_identifiers(
+            chain_state=chain_state, payment_network_id=registry_address
+        )
+        if token_address not in valid_tokens:
+            raise UnknownTokenAddress("Token address is not known in channel settle.")
+
+        channels = views.get_channelstate_settling(
+            chain_state=chain_state,
+            payment_network_id=registry_address,
+            token_address=token_address,
+            creator_address=creator_address
+        )
+
+        channels_to_settle: List[NettingChannelState] = []
+
+        for channel in channels:
+            if channel.partner_state.address == partner_address:
+                channels_to_settle.append(channel)
+
+        return channels
```

Comment:
Be careful here, youre appending to `channels_to_settle `and returning `channels`